vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinServletContextInitializer.java

```diff
@@ -294,9 +296,16 @@ public void failFastContextInitialized(ServletContextEvent event) {
             Stream<Class<? extends Component>> hasErrorComponents = findBySuperType(
                     getErrorParameterPackages(), HasErrorParameter.class)
                             .filter(Component.class::isAssignableFrom)
+                            // Replace Flow default with custom version for Spring
+                            .filter(clazz -> clazz != RouteNotFoundError.class)
                             .map(clazz -> (Class<? extends Component>) clazz);
-            registry.setErrorNavigationTargets(
-                    hasErrorComponents.collect(Collectors.toSet()));
+            final Set<Class<? extends Component>> errorComponents =
+                    hasErrorComponents.collect(Collectors.toSet());
+
+            if (errorComponents.stream().noneMatch(RouteNotFoundError.class::isAssignableFrom)) {
+                errorComponents.add(SpringRouteNotFoundError.class);
```

Comment:
Do I understand correctly, that if the application has a custom `RouteNotFoundError` view (subclass), then this spring view is never shown?

Reply:
I read it like that too, and that is probably how it should be. 
The `SpringRouteNotFoundError` is useful for those who suffer from the Spring dev tools issue initially. I'd suspect that when people add their own error views, they are past the point when they need the instructions. Unless they start by taking a starter like Bakery which overrides the view.

Reply:
Yes, the default message should disappear after being overidden. Bakery could add the same message to its customized view as well if deemed necessary (though probably it isn't).

Reply:
BTW related to the original change, it doesn't anywhere seem to check whether or not the application is in development/production mode. Is it so that the default error view is not even shown in production mode or, is it now outputting the error message even though it is in production mode and where it doesn't make sense for the end user ?

Reply:
Yeah, prod mode should of course be checked 🤦‍♂️. Fixed.